File: backend/src/main/java/com/lawfirm/archive/model/Case.java
package com.lawfirm.archive.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Data
@Entity
@Table(name = "cases")
public class Case {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "case_number")
    private String caseNumber;
    
    private String agents;
    
    @Column(name = "case_reason")
    private String caseReason;
    
    private String court;
    
    @Column(name = "filing_date")
    private LocalDate filingDate;
    
    private String parties;
    
    private String procedure;
    
    private String status;
    
    @Column(name = "category_id")
    private Long categoryId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", insertable = false, updatable = false)
    private Category category;

    @OneToMany(mappedBy = "caseEntity")
    private List<Document> documents = new ArrayList<>();

    @ManyToMany
    @JoinTable(
        name = "case_tags",
        joinColumns = @JoinColumn(name = "case_id"),
        inverseJoinColumns = @JoinColumn(name = "tag_id")
    )
    private Set<Tag> tags = new HashSet<>();
}

File: frontend/src/api/case.ts
import axios from 'axios';

export interface Case {
  id?: number;
  caseNumber: string;
  caseReason: string;
  status: string;
  parties?: string;
  agents?: string;
  filingDate?: string;
  court?: string;
  procedure?: string;
}

const api = axios.create({
  baseURL: 'http://localhost:8080/api'
});

export function getCases() {
  return api.get<Case[]>('/cases');
}

export function createCase(data: Case) {
  return api.post('/cases', data);
}

export function deleteCase(id: number) {
  return api.delete(`/cases/${id}`);
}

export function updateCase(id: number, data: Case) {
  return api.put(`/cases/${id}`, data);
}